tests/mail/default.nix

```
@@ -192,6 +197,24 @@ in
     with subtest("postsuper sudo rule should be present for sudo-srv group"):
       mail.succeed('grep %sudo-srv /etc/sudoers | grep -q postsuper')
 
+    with subtest("imprint redirects to example.com"):
+      client.wait_for_unit("network-online.target")
+
+      mail.wait_for_unit("nginx")
+      mail.wait_for_open_port(80, addr='${nodes.mail.networking.primaryIPAddress}')
+
+      imprint_request_headers = client.succeed("curl mail.example.local -I | xargs -I {} echo {}").split('\n')
```

```suggestion
      imprint_request_headers = client.succeed("curl http://mail.example.local -I | xargs -I {} echo {}").split('\n')
```

Talking about protocol specifications, this makes it a bit more clear, also in case curl defaults ever change.

---

I am also not fully sure why the xargs construction is necessary, doesn't `curl -I` already print headers line by line to stdout by itself?

---

> I am also not fully sure why the xargs construction is necessary, doesn't curl -I already print headers line by line to stdout by itself?

curl doesn't print terminal escape codes when piping it to another command and that way I don't have to bother about dealing with them. If you know a better way (didn't find a useful option in the man page and NO_COLOR=1 doesn't work either), let me know :)

---

As I assume you've done some research and I don't know one either from the top of my head, I'm fine with this hack.

---

I guess piping to `cat` would've worked as well though.